zip-0226.rst

```diff
@@ -1,8 +1,313 @@
+
+The value balances for each Asset identifier in `assetBurn` represents the amount of that Asset identifier that is being burnt. In the case of ZEC, the value balance represents either the transaction fee, or the amount of ZEC changing anonymity pools (to Sapling or Transparent).
+
+Finally, the validator needs to verify the Balance and Binding Signature by adding the value balances for all Assets, as committed using their respective :math:`\mathsf{AssetId}` as the value base point of the Pedersen Commitment. This is done as follows
+
+:math:`\mathsf{bvk = (\sum cv_i^{net})}  - \mathsf{ ValueCommit_0^{Orchard}(v^{balanceOrchard})} - \sum_{\forall \mathsf{AssetId}\textit{ s.t. }\mathsf{v^{AssetId}\neq 0}} \mathsf{ValueCommit_0^{OrchardZSA}(v^{AssetId}, AssetId) } = \sum \mathsf{rcv_{i,j}^{net}}\mathcal{R}^{\mathsf{Orchard}}`
+
+In the case that the balance of all the action values related to a specific Asset will be zero, there will be no value added to the vector. This way, the number of Assets, nor their identifiers will be revealed, except in the case that an Asset is burnt.
+
+**Note:** Even if this mechanism allows having transparent ↔ shielded Asset transfers in theory, the transparent protocol will not be changed with this ZIP to adapt to a multiple Asset structure. This means that unless future consensus rules changes do allow it, unshielding will not be possible for custom Assets.
+
+ZSA Transaction Structure
+=========================
+The transaction format is similar to the version 5 transaction format described in the Zcash specification [#protocol-transactionstructure]_, with the following additions to the Orchard bundle:
+
++-----------------+-------------+-----------------------------------+-------------------------+
+| Bytes           | Name        | Data Type                         | Description             |
++=================+=============+===================================+=========================+
+| varies          | nAssetBurn  | compactSize                       | number of Assets burnt  |
++-----------------+-------------+-----------------------------------+-------------------------+
+| 40*nAssetBurn   | vAssetBurn  | bytes[40][nAssetBurn]             | 32 bytes Asset type_t,  |
+|                 |             |                                   | 8 bytes of valueBalance |
++-----------------+-------------+-----------------------------------+-------------------------+
+
+In terms of the action size, the ZSA action size differs from the Orchard action size by 32 bytes (due to the addition of the :math:`\mathsf{AssetId}`). This implies that the size goes from 820 bytes in the Orchard action to 852 bytes in the ZSA action.
+
+Other Considerations
+====================
+
+Transaction Fees
+----------------
+
+The fee mechanism described in this ZIP will follow the mechanism described in ZIP 317b [#zip-0317b]_.
+
+Security and Privacy
```

Comment:
Top level?

Reply:
Moved Security and Privacy to the top level, yes.